src/storage/restrict.rs

```diff
@@ -0,0 +1,231 @@
+
+use std::marker::PhantomData;
+use std::ops::DerefMut;
+use std::borrow::{Borrow, BorrowMut};
+use std::fmt;
+
+use hibitset::BitSet;
+
+use storage::MaskedStorage;
+use world::EntityIndex;
+use {Component, DistinctStorage, Entities, Entity, Index, Join, Storage, UnprotectedStorage};
+
+/// Generic flags for the restricted storage to determine
+/// the type of restriction.
+pub mod restrict_type {
+    /// Specifies that the `RestrictedStorage` cannot run in parallel.
+    pub struct Normal;
+    /// Specifies that the `RestrictedStorage` can run in parallel.
+    pub struct Parallel;
```

Comment:
I have to prevent `get_mut` on it when you use rayon, since that would mean you could get 2 mutable references to the same entity.